json_schema_fuzz/schema_operations.py

```diff
@@ -1,45 +1,245 @@
 """ Operations on schemas """
 import copy
+import itertools
+import math
 from typing import Any, Dict, List
 
 
+def lcm(numbers):
+    """
+    Find least common multiple of a list of numbers
+
+    TODO replace with math.lcm after updating to Python 3.9
+    """
+    product = 1
+    gcd = 1
+    for num in numbers:
+        gcd = math.gcd(gcd, num)
+        product *= num
+    return product // gcd
+
+
+def get_val_or_none(dictionaries, key):
```

Comment:
We can come up with a more informative name for this. We could mirror the `dict.get` syntax with something like
```python
def get_all(dictionaries, key, *default):
    if len(default) > 1:
        raise TypeError(f"Expected 2 or 3 arguments, received {len(default)}")
    return [
        dictionary[key]
        for dictionary in dictionaries
        if key in dictionary
    ] or default[0]
```